nautobot/circuits/filters.py

```diff
@@ -130,64 +139,59 @@ class CircuitFilterSet(NautobotFilterSet, StatusModelFilterSetMixin, TenancyMode
             "comments": "icontains",
         },
     )
-    provider_id = django_filters.ModelMultipleChoiceFilter(
+    provider = NaturalKeyOrPKMultipleChoiceFilter(
         queryset=Provider.objects.all(),
-        label="Provider (ID)",
+        label="Provider (slug or ID)",
     )
-    provider = django_filters.ModelMultipleChoiceFilter(
-        field_name="provider__slug",
-        queryset=Provider.objects.all(),
-        to_field_name="slug",
-        label="Provider (slug)",
-    )
-    provider_network_id = django_filters.ModelMultipleChoiceFilter(
+    provider_network = NaturalKeyOrPKMultipleChoiceFilter(
         field_name="terminations__provider_network",
         queryset=ProviderNetwork.objects.all(),
-        label="Provider Network (ID)",
-    )
-    type_id = django_filters.ModelMultipleChoiceFilter(
-        queryset=CircuitType.objects.all(),
-        label="Circuit type (ID)",
+        label="Provider Network (slug or ID)",
     )
-    type = django_filters.ModelMultipleChoiceFilter(
-        field_name="type__slug",
+    type = NaturalKeyOrPKMultipleChoiceFilter(
         queryset=CircuitType.objects.all(),
-        to_field_name="slug",
-        label="Circuit type (slug)",
+        label="Circuit type (slug or ID)",
     )
-    site_id = django_filters.ModelMultipleChoiceFilter(
+    site = NaturalKeyOrPKMultipleChoiceFilter(
         field_name="terminations__site",
         queryset=Site.objects.all(),
-        label="Site (ID)",
-    )
-    site = django_filters.ModelMultipleChoiceFilter(
-        field_name="terminations__site__slug",
-        queryset=Site.objects.all(),
-        to_field_name="slug",
-        label="Site (slug)",
+        label="Site (slug or ID)",
     )
     location = TreeNodeMultipleChoiceFilter(
-        field_name="terminations__location__slug",
+        field_name="terminations__location",
         queryset=Location.objects.all(),
-        to_field_name="slug",
         label="Location (slug or ID)",
     )
-    region_id = TreeNodeMultipleChoiceFilter(
-        queryset=Region.objects.all(),
-        field_name="terminations__site__region",
-        label="Region (ID)",
-    )
     region = TreeNodeMultipleChoiceFilter(
         queryset=Region.objects.all(),
         field_name="terminations__site__region",
-        to_field_name="slug",
-        label="Region (slug)",
+        label="Region (slug or ID)",
+    )
+    has_terminations = RelatedMembershipBooleanFilter(
+        field_name="terminations",
+        label="Has terminations",
+    )
+    termination_a = django_filters.ModelMultipleChoiceFilter(
+        queryset=CircuitTermination.objects.all(),
+        label="Termination A (ID)",
+    )
+    termination_z = django_filters.ModelMultipleChoiceFilter(
+        queryset=CircuitTermination.objects.all(),
+        label="Termination Z (ID)",
     )
-    tag = TagFilter()
+    tags = TagFilter()
```

Comment:
Could we not just map `TaggableManager` to `TagFilter` in `BaseFilterSet` and not have to hard-code this on every `FilterSet`?

Reply:
Nevermind. I saw #3154